src/upgrade/steps/binary-cache-flush.js

```
@@ -0,0 +1,84 @@
+  const marker = readVersionMarker(cacheDir);
+  const binaryExists = fs.existsSync(path.join(cacheDir, 'bin', 'postgres'));
+
+  if (binaryExists && marker === pinned) {
+    return { status: 'SKIP', detail: `binary OK (PG ${pinned} matches marker)` };
+  }
+
+  // Delegate to existing postgres.js downloadBinary if available; else fail-loud
+  // and instruct operator. We avoid duplicating the download logic here.
+  let downloadFn;
+  try {
+    const postgres = require('../../postgres');
```

ESLint

[error] 'require' is not defined.

(no-undef)